apps/survey/app/src/script.js

```diff
@@ -108,11 +111,30 @@ async function createStore(token, tokenSettings) {
             nextState = await startSurvey(nextState, returnValues)
             break
           case 'CastVote':
-            nextState = await castVote(
+            nextState = await processVote(
               nextState,
               returnValues,
               blockNumber,
-              transactionIndex
+              transactionIndex,
+              logIndex
+            )
+            break
+          case 'ResetVote':
+            // We have to manually calculate the option power left when a vote is reset
+            returnValues = {
+              ...returnValues,
+              // TODO: use BN.js instead
+              optionPower: String(
```

Comment:
This is not really nice, but I didn't want to deal with the hassle of adding BN to this script (it should do all math via BN) until we completely refactor `bignumber` to `BN.js` in this app.

Reply:
[bignumber.js](https://github.com/MikeMcl/bignumber.js/) allows doing arithmetic and also recommends instantiating with a string for numbers of limited precision. Which problem do you aim so solve by switching to `BN.js`?

Reply:
`BN.js` is faster and more widely used, compared to `bignumber.js`. This app should eventually be converted to completely use `BN.js` for all math operations in the future.